File: src/services/types/command-center.ts
export interface CommandCenterKPIs {
  total_members: number;
  avg_risk_score: number | null;
  hba1c_control_rate: number | null;
  open_care_gaps: number;
  pdc_above_80_rate: number | null;
}

export interface ActionQueueItem {
  id: string;
  patient_id: string;
  patient_name: string;
  template_id: string;
  program_id: string;
  cohort_id: string | null;
  cohort_name: string | null;
  priority: number;
  title: string;
  description: string | null;
  status: string;
  assigned_to: string | null;
  resolution_options: Array<{
    label: string;
    action_type: string;
    icon?: string;
    channel?: string;
    template_slug?: string;
    navigate_to?: string;
    navigate_tab?: string;
    requires_reason?: boolean;
  }>;
  trigger_data: Record<string, unknown> | null;
  created_at: string;
}

export interface ActionQueueResponse {
  items: ActionQueueItem[];
  total: number;
}

export interface AIInsightsResponse {
  markdown: string;
  generated_at: string;
  is_cached: boolean;
}

export interface UpcomingReviewItem {
  patient_id: string;
  patient_name: string;
  program_id: string;
  program_name: string;
  cohort_name: string;
  cohort_color: string;
  review_due_at: string;
  days_until_due: number;
}

export interface UpcomingReviewsResponse {
  items: UpcomingReviewItem[];
  total: number;
}

File: backend/app/schemas/command_center.py
"""Pydantic schemas for command center endpoints."""

from __future__ import annotations

from pydantic import BaseModel


# ── KPIs ──────────────────────────────────────────────────────────────────

class CommandCenterKPIs(BaseModel):
    total_members: int
    avg_risk_score: float | None
    hba1c_control_rate: float | None
    open_care_gaps: int
    pdc_above_80_rate: float | None


# ── Action Queue ──────────────────────────────────────────────────────────

class ActionChip(BaseModel):
    label: str
    action_type: str
    target: str


class ActionQueueItem(BaseModel):
    id: str
    patient_id: str
    patient_name: str
    alert_type: str
    title: str
    description: str
    priority: str
    cohort_name: str
    cohort_color: str
    actions: list[ActionChip]


class ActionQueueResponse(BaseModel):
    items: list[ActionQueueItem]
    total: int


# ── AI Insights ───────────────────────────────────────────────────────────

class AIInsightsResponse(BaseModel):
    markdown: str
    generated_at: str
    is_cached: bool


# ── Upcoming Reviews ──────────────────────────────────────────────────────

class UpcomingReviewItem(BaseModel):
    patient_id: str
    patient_name: str
    program_id: str
    program_name: str
    cohort_name: str
    cohort_color: str
    review_due_at: str
    days_until_due: int


class UpcomingReviewsResponse(BaseModel):
    items: list[UpcomingReviewItem]
    total: int
